test_create_garden.py

```diff
@@ -0,0 +1,113 @@
+            ["W", ".", ".", ".", ".", ".", ".", ".", ".", "F"],
+            ["F", ".", "W", ".", ".", "W", ".", ".", "F"],
+            [".", "W", ".", ".", ".", ".", ".", "W", "."],
+            [".", "F", ".", ".", ".", ".", ".", ".", ".", "."],
+            [".", "F", ".", ".", ".", ".", ".", ".", "F"],
+            [".", ".", "F", ".", ".", "W", "F", ".", "F"],
+            [".", ".", "W", ".", ".", "W", ".", ".", "."],
+        ]
+        self.garden2 = [
+            [".", ".", ".", ".", ".", ".", ".", ".", "."],
+            [".", ".", ".", ".", ".", ".", ".", ".", "."],
+            [".", ".", ".", ".", ".", ".", ".", ".", "."],
+            [".", ".", ".", ".", ".", ".", ".", ".", "."],
+            [".", ".", ".", ".", ".", ".", ".", ".", "."],
+            [".", ".", ".", ".", ".", ".", ".", ".", "."],
+            [".", ".", ".", ".", ".", ".", ".", ".", "."],
+        ]
+
+        # Number of flowers (F): 0
+        # Number of weeds (W): 0
+
+        self.garden3 = [
+            ["F", "F", "F", "F", "F", "F", "F", "F", "F"],
+            ["F", "F", "F", "F", "F", "F", "F", "F", "F"],
+            ["F", "F", "F", "F", "F", "F", "F", "F", "F"],
+            ["F", "F", "F", "F", "F", "F", "F", "F", "F"],
+            ["F", "F", "F", "F", "F", "F", "F", "F", "F"],
+            ["F", "F", "F", "F", "F", "F", "F", "F", "F"],
+            ["F", "F", "F", "F", "F", "F", "F", "F", "F"],
+        ]
+        # Number of flowers (F): 63
+        # Number of weeds (W): 0
+
+        self.garden4 = [
+            ["W", "W", "W", "W", "W", "W", "W", "W", "W"],
+            ["W", "W", "W", "W", "W", "W", "W", "W", "W"],
+            ["W", "W", "W", "W", "W", "W", "W", "W", "W"],
+            ["W", "W", "W", "W", "W", "W", "W", "W", "W"],
+            ["W", "W", "W", "W", "W", "W", "W", "W", "W"],
+            ["W", "W", "W", "W", "W", "W", "W", "W", "W"],
+            ["W", "W", "W", "W", "W", "W", "W", "W", "W"],
+        ]
+        # Number of flowers (F): 0
+        # Number of weeds (W): 63
+
+        self.garden5 = [
+            ["f", ".", ".", ".", ".", ".", ".", "w", "."],
+            [".", ".", ".", ".", "w", ".", ".", ".", "."],
+            [".", ".", ".", ".", ".", ".", "f", ".", "."],
+            [".", "f", ".", ".", ".", ".", ".", ".", "."],
+            [".", ".", ".", ".", ".", ".", ".", "w", "."],
+            [".", ".", ".", "f", ".", ".", ".", ".", "."],
+            [".", ".", ".", ".", ".", "w", ".", ".", "."],
+        ]
+
+    def test_create_func(self):
+        test_case_1 = [-1, -1]
+        test_case_2 = [100, 100]
+        test_case_3 = [7, 7]
+        test_case_4 = [2.25, 3.456]
+
+        self.assertRaises(ValueError, create_garden, test_case_1[0], test_case_1[1])
+        self.assertEqual(len(create_garden(test_case_2[0], test_case_2[1])), 100)
+        self.assertEqual(len(create_garden(test_case_3[0], test_case_3[1])), 7)
+        self.assertRaises(ValueError, create_garden, test_case_4[0], test_case_4[1])
```

Comment:
Where are we testing that flowers/weeds are being planted randomly and by chance?

Reply:
here i am testing the total rows and colummns in the program and validating them. Can't do them anywher else because they are inputted in the main code

Reply:
Please read the review carefully. Also, where is the number of columns tested?